src/wallet3/transaction_constructor.cpp

```diff
@@ -1,13 +1,103 @@
 #include "transaction_constructor.hpp"
 #include "pending_transaction.hpp"
+#include <sqlitedb/database.hpp>
 
 namespace wallet
 {
   PendingTransaction
-  TransactionConstructor::CreateTransaction(
-      const std::vector<std::pair<address, uint64_t>>& recipients, uint64_t feePerKB) const
+  TransactionConstructor::CreateTransaction(const std::vector<TransactionRecipient>& recipients, int64_t feePerKB) const
   {
-    return {};
+    PendingTransaction txNew(recipients);
+    SelectInputsAndFinalise(txNew);
+    return txNew;
+  }
+  
+  // SelectInputs will choose some available unspent outputs from the database and allocate to the transaction
+  // can be called multiple times and will add until enough is sufficient
+  void
+  TransactionConstructor::SelectInputs(PendingTransaction& ptx) const
+  {
+    // Fail early
+    int64_t estimated_fee = EstimateFee();
+    //int64_t estimated_fee = estimate_fee(2, fake_outs_count, min_outputs, extra.size(), clsag, base_fee, fee_percent, fixed_fee,     fee_quantization_mask);
+    int64_t transaction_total = ptx.SumOutputs() + estimated_fee;
+    int64_t wallet_balance = db->prepared_get<int>("SELECT sum(amount) FROM outputs");
+
+    // Check that we actually have enough in the outputs to build this transaction.
+    if (wallet_balance < transaction_total)
+      throw std::runtime_error("Insufficient Wallet Balance");
+
+
+    int64_t shortfall = transaction_total - ptx.SumInputs();
+
+    // If we already have enough inputs return
+    if (shortfall < 0)
+      return;
+
+    // Prefer a single output if suitable
+    SQLite::Statement single_output{db->db, "SELECT * FROM outputs WHERE amount > ? ORDER BY amount ASC LIMIT 1"};
```

Comment:
Prefer `SELECT cola, colb, colc` over `SELECT *` because, in the future, we may need to make database migrations that add/remove columns, but that will result in different column *orders* between new and upgraded versions.  Whereas if you explicitly list columns that can't happen.